tests/unit/optimization/StanStochasticMinimizers_unittest.cc

```diff
@@ -0,0 +1,167 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Elfarouk, Wu Lin
+ */
+
+#include <shogun/optimization/StanFirstOrderSAGCostFunction.h>
+#include <gtest/gtest.h>
+#include "StanStochasticMinimizers_unittest.h"
+#include <shogun/optimization/L2Penalty.h>
+#include <shogun/optimization/SGDMinimizer.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/optimization/SVRGMinimizer.h>
+#include <shogun/base/Parameter.h>
+#include <shogun/lib/Map.h>
+#include <shogun/optimization/StandardMomentumCorrection.h>
+#include <shogun/optimization/AdaDeltaUpdater.h>
+#include <shogun/optimization/AdamUpdater.h>
+#include <shogun/optimization/NesterovMomentumCorrection.h>
+#include <shogun/optimization/RmsPropUpdater.h>
+#include <shogun/optimization/AdaptMomentumCorrection.h>
+#include <shogun/optimization/L1PenaltyForTG.h>
+#include <shogun/optimization/InverseScalingLearningRate.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/SMIDASMinimizer.h>
+#include <shogun/optimization/PNormMappingFunction.h>
+using namespace shogun;
+using namespace Eigen;
+using namespace stan::math;
+using std::function;
+
+StanVector generateParameters()
+{
+	StanVector w(3, 1);
+	var x1(0), x2(0), x3(0);
+	w(0, 0) = x1;
+	w(1, 0) = x2;
+	w(2, 0) = x3;
+	return w;
+}
+
+function<var(const StanVector&, float64_t)>
+cost_for_ith_datapoints(SGMatrix<float64_t>& X, SGMatrix<float64_t>& y)
```

Comment:
Refactor.

Reply:
Will be refactoring the stan tests once the neural network is working